create-ao-dapp/templates/scripts/check_deps-lua.sh

```diff
@@ -8,9 +8,19 @@ RED='\033[0;31m'    # Red color for errors
 GREEN='\033[0;32m'  # Green color for success messages
 NC='\033[0m'        # No Color (reset to default)
 
-# Check if Lua is installed and the correct version
-if ! lua -v | grep -q "$LUA_VERSION"; then
-    echo -e "${RED}Lua $LUA_VERSION is not installed.${NC}"
+# Function to compare version strings
+version_greater_equal() {
+    if [ "$(printf '%s\n' "$2" "$1" | sort -V | head -n1)" = "$2" ]; then
+        return 0
+    else
+        return 1
+    fi
+}
+
+# Check if Lua is installed and version is >= 5.4.6
+INSTALLED_VERSION=$(lua -v 2>&1 | grep -o '[0-9]\+\.[0-9]\+\.[0-9]\+')
```

Comment:
You can use `-E` for grep to avoid escaping here:

`INSTALLED_VERSION="$(lua -v 2>&1 | grep -oE '[0-9]+\.[0-9]+\.[0-9]+')"`